### PR TITLE
[5.5.0.RC2]Upgrade to spring io platform Cairo-SR1. #810

### PR DESCRIPTION
**Pull request by btournc**
Please review #810
1. Change version to spring io platform Cairo-SR1
------------------------------------------------------

・terasoluna-gfw-parent/pom.xml 
<io.spring.platform.version> : Cairo-SR1
<jadira-usertype-core.version> : 6.0.1.GA
java-version : 1.8
<cargo.tomcat9.version> : 9.0.6

・terasoluna-gfw-common-libraries/pom.xml 
java-version : 1.8

2. Fix build error of UriUtils
------------------------------------------------------

・Error contents
[ERROR] /home/travis/build/terasolunaorg/terasoluna-gfw/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java:[180,11] exception java.io.UnsupportedEncodingException is never thrown in body of corresponding try statement

https://travis-ci.org/terasolunaorg/terasoluna-gfw/jobs/391071386#L2025

・Error occurrence location
https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java#L180

• Error cause
With Spring 5, UriUtils.encodeQueryParam has changed to not throw UnsupportedEncodingException.

https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/UriUtils.html#encodeQueryParam-java.lang.String-java.nio.charset.Charset-

· Correspondence to errors
Modify not to catch UnsupportedEncodingException.

3. Fix build error of Mockito
------------------------------------------------------

・Error contents
[ERROR] /home/travis/build/terasolunaorg/terasoluna-gfw/terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/test/java/org/terasoluna/gfw/common/exception/ExceptionLoggerTest.java:[776,53] <anonymous org.terasoluna.gfw.common.exception.ExceptionLoggerTest$1> is not abstract and does not override abstract method matches(java.lang.Object) in org.mockito.ArgumentMatcher
https://travis-ci.org/terasolunaorg/terasoluna-gfw/jobs/391074725#L931

・Error occurrence location
https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/test/java/org/terasoluna/gfw/common/exception/ExceptionLoggerTest.java#L778

https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java#L573

https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/exception/HandlerExceptionResolverLoggingInterceptorTest.java#L545

https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/logging/HttpSessionEventLoggingListenerTest.java#L282

• Error cause
This API was changed in Mockito 2.1.0 
matches(Object argument) -> matches(T argument)

https://static.javadoc.io/org.mockito/mockito-core/2.2.9/org/mockito/ArgumentMatcher.html

· Correspondence to errors
Modify as follows:
matches(Object argument)  -> matches(LoggingEvent argument) 

4. Fix failed tests of Mockito
------------------------------------------------------

・Error contents
CompositeRequestDataValueProcessorTest.testProcessActionSameActionAndResult01:64 
Expected: is "action"
     but: was null
  CompositeRequestDataValueProcessorTest.testProcessActionDifferectActionAndResult01:80 
Expected: is "other_action"
     but: was null

https://travis-ci.org/terasolunaorg/terasoluna-gfw/jobs/391107378#L1791

・Error occurrence location
https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java#L60

• Error cause
Since Mockito 2.1.0, anyString() is only allow non-null String.
 
https://static.javadoc.io/org.mockito/mockito-core/2.18.3/org/mockito/ArgumentMatchers.html#anyString--

· Correspondence to errors
Depending on the source below, method parameter is fixed as "null".
anyString () -> Mockito.isNull ()

https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessor.java#L84

5. Fix failed tests of Spring
------------------------------------------------------

・Error contents
FunctionsTest.testU_EncodingDelimiter:320 
Expected: is "%2B"
     but: was "+"
  FunctionsTest.testQuery04_url_encoding_for_queryString:545 
Expected: is "key%231=value%231&key%5B2=value%5B2&key%5D3=value%5D3&key%264=value%264&key%3D5=value%3D5&key%2B6=value%2B6"
     but: was "key%231=value%231&key%5B2=value%5B2&key%5D3=value%5D3&key%264=value%264&key%3D5=value%3D5&key+6=value+6"

https://travis-ci.org/terasolunaorg/terasoluna-gfw/jobs/391107378#L1797

・Error occurrence location
https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/el/FunctionsTest.java#L320

• Error cause
In Spring 4, +(0x2b) is encoding to %2B, But Since Spring 5, it does not encode to %2B.

https://jira.spring.io/browse/SPR-14828

· Correspondence to errors
Change %2B to +.

6. Fix build error of JDK7
------------------------------------------------------

・Error contents
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project terasoluna-gfw-common: Fatal error compiling: invalid target release: 1.8 

https://travis-ci.org/terasolunaorg/terasoluna-gfw/jobs/391551319#L830

・Error occurrence location
https://github.com/terasolunaorg/terasoluna-gfw/blob/5.5.0.RC2-development/.travis.yml#L6

• Error cause
Java 7 will not be supported from this term development version.

· Correspondence to errors
Delete openjdk7.

### CODE REVIEW

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java [hunk 1]
```diff
@@ -51,14 +52,12 @@ public void setup() {
...
-        // for Spring3
         when(requestDataValueProcessor.processAction(
                 (HttpServletRequest) (anyObject()), anyString())).thenReturn(
                         "action");
-        // for Spring4
         when(requestDataValueProcessor.processAction(
-                (HttpServletRequest) (anyObject()), anyString(), anyString()))
-                        .thenReturn("action");
+                (HttpServletRequest) (anyObject()), anyString(), Mockito
+                        .isNull())).thenReturn("action");
```
Comment (yoshikawaa) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: - [ ] It's better `ArgumentMatchers#nullable` than `Mockito#isNull`. And please use static import.

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/el/FunctionsTest.java [hunk 2]
```diff
@@ -317,7 +317,7 @@ public void testU_DIGIT() {
 
     @Test
     public void testU_EncodingDelimiter() {
-        assertThat(Functions.u("+"), is("%2B"));
+        assertThat(Functions.u("+"), is("+"));
```
Comment (yoshikawaa) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/el/FunctionsTest.java]: We re-consider this problem at #757 .

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java [hunk 3]
```diff
@@ -175,14 +175,7 @@ public static String u(String value) {
         if (value == null || value.isEmpty()) {
             return "";
         }
-        try {
-            return UriUtils.encodeQueryParam(value, "UTF-8");
-        } catch (UnsupportedEncodingException ignored) {
```
Comment (yoshikawaa) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java]: - [ ] Oh... Please remove `import java.io.UnsupportedEncodingException;`.

File: terasoluna-gfw-parent/pom.xml [hunk 4]
```diff
@@ -392,28 +392,28 @@
...
     <cargo.daemon.url>http://localhost:18000</cargo.daemon.url>
-    <cargo.container.id>tomcat8x</cargo.container.id>
-    <cargo.container.url>http://archive.apache.org/dist/tomcat/tomcat-8/v${cargo.tomcat8.version}/bin/apache-tomcat-${cargo.tomcat8.version}.zip</cargo.container.url>
-    <cargo.tomcat8.version>8.5.20</cargo.tomcat8.version>
+    <cargo.container.id>tomcat9x</cargo.container.id>
+    <cargo.container.url>http://archive.apache.org/dist/tomcat/tomcat-9/v${cargo.tomcat9.version}/bin/apache-tomcat-${cargo.tomcat9.version}.zip</cargo.container.url>
+    <cargo.tomcat9.version>9.0.6</cargo.tomcat9.version>
   </properties>
   <profiles>
     <profile>
```
Comment (yoshikawaa) [inline terasoluna-gfw-parent/pom.xml]: (Other Point)

https://github.com/terasolunaorg/terasoluna-gfw/blob/68533fd2fa2216359cebc88f3a14c15b065c3db7/terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/pagination/PaginationInfoTest.java#L65

- [ ] `PageRequest(int page, int size)` is deprecated constructor.

Comment (yoshikawaa) [inline terasoluna-gfw-parent/pom.xml]: (Other Point)
https://github.com/terasolunaorg/terasoluna-gfw/blob/68533fd2fa2216359cebc88f3a14c15b065c3db7/terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/pagination/PaginationTagTest.java#L415

- [ ] `Sort(Order... orders)` is deprecated constructor.

Comment (yoshikawaa) [inline terasoluna-gfw-parent/pom.xml]: In addition to what I reviewed out, there are a lot of warnings by the IDE.
We should create another Issue and respond.

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java [hunk 5]
```diff
@@ -51,14 +52,12 @@ public void setup() {
...
         // Set mock behavior
-        // for Spring3
         when(requestDataValueProcessor.processAction(
                 (HttpServletRequest) (anyObject()), anyString())).thenReturn(
                         "action");
-        // for Spring4
         when(requestDataValueProcessor.processAction(
-                (HttpServletRequest) (anyObject()), anyString(), anyString()))
-                        .thenReturn("action");
+                (HttpServletRequest) (anyObject()), anyString(), Mockito
```
[the remark this answers is not in the document]
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: Change anyObject () to any ().

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java (hunk 1 above)
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: Changed Mockito#isNull to ArgumentMatchers#nullable.

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/el/FunctionsTest.java (hunk 2 above)
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/el/FunctionsTest.java]: ok

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java (hunk 3 above)
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java]: It has been deleted.

File: terasoluna-gfw-parent/pom.xml (hunk 4 above)
Comment (btournc) [reply terasoluna-gfw-parent/pom.xml]: Changed new PageRequest() to PageRequest.of().

Comment (btournc) [reply terasoluna-gfw-parent/pom.xml]: Changed new Sort() to Sort.by().

Comment (btournc) [reply terasoluna-gfw-parent/pom.xml]: ok

File: terasoluna-gfw-parent/pom.xml [hunk 6]
```diff
@@ -392,28 +392,28 @@
...
+    <java-version>1.8</java-version>
     <!-- == Cargo Plugin == -->
     <cargo.jvmargs>-Xms512m -Xmx1024m</cargo.jvmargs>
     <cargo.daemon.url>http://localhost:18000</cargo.daemon.url>
-    <cargo.container.id>tomcat8x</cargo.container.id>
-    <cargo.container.url>http://archive.apache.org/dist/tomcat/tomcat-8/v${cargo.tomcat8.version}/bin/apache-tomcat-${cargo.tomcat8.version}.zip</cargo.container.url>
-    <cargo.tomcat8.version>8.5.20</cargo.tomcat8.version>
+    <cargo.container.id>tomcat9x</cargo.container.id>
+    <cargo.container.url>http://archive.apache.org/dist/tomcat/tomcat-9/v${cargo.tomcat9.version}/bin/apache-tomcat-${cargo.tomcat9.version}.zip</cargo.container.url>
+    <cargo.tomcat9.version>9.0.6</cargo.tomcat9.version>
```
Comment (hirakuriy) [inline terasoluna-gfw-parent/pom.xml]: - [ ] Due to Tomcat 9.0.6's bug, we decided to use Tomcat 9.0.5.  Please change the version from 9.0.6 to 9.0.5 .

Comment (btournc) [reply terasoluna-gfw-parent/pom.xml]: @hirakuriy 
 Fixed on below:
 f08e68c72e7f91d94a60ca7105ac438041dcc800

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (hirakuriy) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: @btournc 
`anyObject()` was deprecated according to Spring IO Platform.  So you have to modify these method in this issue.

[the remark this answers is not in the document]
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: @hirakuriy 
The source CompositeRequestDataValueProcessorTest.java is already fixed with below:  
d5bb988b0f3341ce81324e327b3194a130471c9f 

The following sources are fixed with below: 
ab60a1af2c1a9818d787646f55940a4fe616b250 
ExceptionLoggerTest.java
MessagesPanelTagTest.java
ResultMessagesLoggingInterceptorTest.java
TransactionTokenInterceptorTest.java

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java [hunk 7]
```diff
@@ -37,7 +34,8 @@
  * <ul>
  * <li>Escaping HTML tag using {@code f:h}</li>
  * <li>Encoding URL using {@code f:u}</li>
- * <li>Replacing new line characters with {@code <br />} using {@code f:br}</li>
+ * <li>Replacing new line characters with {@code <br />
+ * } using {@code f:br}</li>
```
Comment (hirakuriy) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java]: @btournc
- [ ] Please make it back to the original.

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java [hunk 8]
```diff
@@ -419,7 +419,8 @@ public void testDoStartTagInternalPanelElementEmpty() throws Exception {
     }
 
     /**
-     * Set default messages attribute name & Use ResultMessages & change PanelElement,OuterElement and InnerElement is empty.<br>
+     * Set default messages attribute name & Use ResultMessages & change PanelElement,OuterElement and InnerElement is
+     * empty.<br>
```
Comment (hirakuriy) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java]: @btournc 
- [ ] Please make it back to the original.

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (hirakuriy) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: OK

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java (hunk 1 above)
Comment (hirakuriy) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/mvc/support/CompositeRequestDataValueProcessorTest.java]: OK

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java (hunk 3 above)
Comment (hirakuriy) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java]: OK

File: terasoluna-gfw-parent/pom.xml (hunk 4 above)
Comment (hirakuriy) [reply terasoluna-gfw-parent/pom.xml]: OK

Comment (hirakuriy) [reply terasoluna-gfw-parent/pom.xml]: OK

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java (hunk 7 above)
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java]: @hirakuriy
 Fixed on below:
 282ea5b53d12a7ca68c5ee75754d69edc920e1c0

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java (hunk 8 above)
Comment (btournc) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java]: @hirakuriy
 Fixed on below:
 282ea5b53d12a7ca68c5ee75754d69edc920e1c0

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java (hunk 7 above)
Comment (hirakuriy) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/el/Functions.java]: OK

File: terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java (hunk 8 above)
Comment (hirakuriy) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java]: OK

**Pull request merged**